osu.Game/Screens/Play/Player.cs

```diff
@@ -197,6 +209,13 @@ private void load(AudioManager audio, OsuConfigManager config)
                 skipOverlay.Hide();
             }
 
+            if (game != null)
+                OverlayActivationMode.BindTo(game.OverlayActivationMode);
+
+            gameplayOverlaysDisabled.BindValueChanged(_ => updateOverlayActivationMode());
+            DrawableRuleset.IsPaused.BindValueChanged(_ => updateOverlayActivationMode());
+            DrawableRuleset.HasReplayLoaded.BindValueChanged(_ => updateOverlayActivationMode());
```

Comment:
sounds like we might want to take a lease on the bindable instead of directly binding

Reply:
oh just realized I didn't mention this, but the settings overlay closing is the result of me setting `Disable overlays during gameplay` on, I'm suggesting to move these binds inside `OnEntering()` where it makes sense to start allowing to change the `OverlayActivationMode`.

Reply:
Alternatively, adding a check that `Player` is the current screen should do the trick aswell.

Reply:
Oh, now that you mention it, this'll also change `OverlayActivationMode` on the results screen, tbh I think moving all this overlay activation mode logic into an `OsuGame` level would be a better choice as I mentioned above https://github.com/ppy/osu/pull/9766#pullrequestreview-463824776 (in a separate PR this would depend on)

Reply:
taking a lease on the bindable would also work as desired by rolling the changes back, and require no other changes.

Reply:
> 
> 
> Oh, now that you mention it, this also changes `OverlayActivationMode` on the results screen, tbh I think moving all this overlay activation mode logic into an `OsuGame` level would be a better choice as I mentioned above [#9766 (review)](https://github.com/ppy/osu/pull/9766#pullrequestreview-463824776) (in a separate PR this would depend on)
> 
> Edit: Adding the `Player` check would fix this as well but it would just make the whole thing look like it shouldn't be there.

Did some debugging as this sounded quite weird to me. Turns out that the the behaviour described in results screen is
because of this: https://github.com/ppy/osu/blob/9a09f97478f063384a03c0d35d23f359cc9d1353/osu.Game/Screens/Ranking/ResultsScreen.cs#L33

![image](https://user-images.githubusercontent.com/20256717/89809612-111f3500-db3c-11ea-8ea6-093d1e36b665.png)

As seen on the screen, the activation mode is `All`, which is the default activation mode of an `OsuScreen`. Would've been either `UserTriggered` or `Disabled` if  `Player` was to change the `OverlayActivationMode`.